### PR TITLE
Added subtitle support to vidmoly resolver

### PR DESCRIPTION
**Pull request by MrDini123**
Sadly not a full sub to test, but if you seek to ~36 seconds, you should see some captions: `https://vidmoly.net/c371kucefss1.html`

### CODE REVIEW

File: script.module.resolveurl/lib/resolveurl/plugins/vidmoly.py [hunk 1]
```diff
@@ -16,22 +16,36 @@
...
-    def get_media_url(self, host, media_id):
-        return helpers.get_media_url(
-            self.get_url(host, media_id),
-            patterns=[r'''sources:\s*\[{file:"(?P<url>[^"]+)'''],
-            result_blacklist=['.mpd'],
-            referer=True
-        )
+    def get_media_url(self, host, media_id, subs=False):
+        web_url = self.get_url(host, media_id)
+        self.headers.update({'Referer': web_url, 'Origin': urllib_parse.urljoin(web_url, '/')})
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/vidmoly.py:34]: Origin should not have a trailing '/' sho add a [:-1]
`'Origin': urllib_parse.urljoin(web_url, '/')[:-1]`

**Review by Gujal00 — CHANGES_REQUESTED** [review]

File: script.module.resolveurl/lib/resolveurl/plugins/vidmoly.py (hunk 1 above)
Comment (MrDini123) [reply script.module.resolveurl/lib/resolveurl/plugins/vidmoly.py]: Right, done.

**Comment by Gujal00** [comment]
i merged the dood one but it is returning irrelevant stuff as subs, will fix that and then look at adding a mechanism of subs to generic resolver as you have now found two sources with subs that are currently using generic resolver. Will keep this on hold for now

**Comment by MrDini123** [comment]
Thanks!

I am sad to hear that. I personally haven't encountered such a case. That's why I kept the `kind: captions` inside the regex...

**Comment by Gujal00** [comment]
@MrDini123 please pull the latest commits and test. I have added subtitle support to the Generic resolver

**Comment by MrDini123** [comment]
@Gujal00 You are quick as always. Thanks a bunch! Works flawlessly. All three recently touched providers display the subtitles.

Although I gotta admit that I don't quite understand these two lines yet:

https://github.com/Gujal00/ResolveURL/blob/bb15c19a25a400e02834ab52d475af0342d1c0db/script.module.resolveurl/lib/resolveurl/lib/helpers.py#L247-L248

I assumed that you will introduce a `subtitle_pattern` kwarg in `get_media_url` similarly to how `patterns` is done, but this way if I want to add subtitle support to a resolver that uses the generic resolver, do I have to always edit helpers with the new regexp?

**Comment by Gujal00** [comment]
> @Gujal00 You are quick as always. Thanks a bunch! Works flawlessly. All three recently touched providers display the subtitles.
> I assumed that you will introduce a `subtitle_pattern` kwarg in `get_media_url` similarly to how `patterns` is done, but this way if I want to add subtitle support to a resolver that uses the generic resolver, do I have to always edit helpers with the new regexp?

Yes there is kwargs introduced for this, generic resolver always has some generic patterns already and you only need to provide your own pattern if it is not working with the generic patterns, For now the two seemed fairly generic so i made them generic patterns.

The kwargs supported by the Generic Resolver are here. `subs` `subs_patterns` `generic_subs_patterns`
https://github.com/Gujal00/ResolveURL/blob/bb15c19a25a400e02834ab52d475af0342d1c0db/script.module.resolveurl/lib/resolveurl/lib/helpers.py#L256-L260

**Comment by MrDini123** [comment]
I see, then its perfect. Thanks once again. Gonna close these two PRs.

**Pull request closed without merging**